### PR TITLE
feat: PPOM version update to fetch files only when transaction is received

### PR DESCRIPTION
**Pull request by jpuri**
## **Description**

Blockaid - download files from CDN only when transaction is received and we do not have updated files in storage.

## **Related issues**

Fixes: MetaMask/MetaMask-planning#2110

## **Manual testing steps**

1. To to test dapp
2. Submit transaction and check file downloads
3. Check by switching between various networks, submitting multiple transactions.

## **Screenshots/Recordings**
NA

## **Pre-merge author checklist**

- [X] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [X] I've clearly explained what problem this PR is solving and how it is solved.
- [X] I've linked related issues
- [X] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [X] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [X] I’ve properly set the pull request status:
  - [X] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by jpuri** [comment]
@SocketSecurity ignore npm/@metamask/ppom-validator@0.25.0

**Review by naugtur — CHANGES_REQUESTED** [review]
blocking update to ppom-validator v0.25.0 
Feel free to override after we've had a conversation or if we've ran out of time.

**Review by naugtur — DISMISSED** [review]

**Comment by naugtur** [comment]
I'm gonna unblock this PR before I go offline (the only way to do it myself was to approve, sorry) but I would like to handle the privacy impact at least by surfacing to the user that something is being sent.

**Comment by seaona** [comment]
hi there :wave: from QA I see the following behaviour, which I think it is expected:
- switching to a new network does not trigger any CDN request
- starting a transaction/signature triggers a `HEAD` request
  - if it's the 1st transaction/signature in that network (there is no data in the ppom-db for that net), the subsequent cdn requests are made to get the config, stale and stale_diff data
  - if it's not the 1st transaction in that network, and there are no updates, no subsequent cdn requests are made

https://github.com/MetaMask/metamask-extension/assets/54408225/da6380de-b887-4bb7-8fd2-5c9365ff6c31

https://github.com/MetaMask/metamask-extension/assets/54408225/b11d370f-c866-4182-98d6-657156e21575


@jpuri let me know if this sounds good too :pray:

**Comment by seaona** [comment]
sounds good @jpuri I can do that and report back :+1:

**Review by digiwand — DISMISSED** [review]

**Review by blackdevelopa — DISMISSED** [review]

**Comment by jpuri** [comment]
@metamaskbot update-policies

**Comment by jpuri** [comment]
@SocketSecurity ignore npm/@metamask/ppom-validator@0.26.0

**Comment by jpuri** [comment]
@metamaskbot update-policies

File: ui/pages/confirmations/components/confirm/info/info copy.tsx
Comment (segun) [inline ui/pages/confirmations/components/confirm/info/info copy.tsx:1]: Is this change supposed to be part of this PR?

**Review by segun — DISMISSED** [review]

**Review by blackdevelopa — DISMISSED** [review]

**Review by segun — APPROVED** [review]

**Review by blackdevelopa — APPROVED** [review]

**Pull request merged**